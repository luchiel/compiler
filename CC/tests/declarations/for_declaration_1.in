int main()
{
int n = 6, m = 5;
for(int i = 0; i < n; ++i);
}