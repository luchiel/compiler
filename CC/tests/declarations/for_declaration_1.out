type <int>
type <float>
type <void>
type <int*>
	pointer to <int>
type <void*>
	pointer to <void>
variable NULL of type <void*> initialized by
	{int 0}
		type <int>
function main
	returns type <int>
	arguments
		<no args>
	{{}}
		variable n of type <int> initialized by
			{int 6}
				type <int>
		variable m of type <int> initialized by
			{int 5}
				type <int>
	| 
	+-{for}
		variable i of type <int> initialized by
			{int 0}
				type <int>
	  | 
	  +-{;}
	  | | 
	  | +-{<}
					type <int>
	  |   | 
	  |   +-{i}
						type <int>
	  |   | 
	  |   +-{n}
						type <int>
	  | 
	  +-{++x}
				type <int>
	  | | 
	  | +-{i}
					type <int>
	  | 
	  +-{;}
